lib/pbench/agent/results.py

```diff
@@ -18,49 +17,61 @@ def __init__(self, result_dir, target_dir, user, prefix, config, logger):
         assert (
             config and logger
         ), f"config, '{config!r}', and/or logger, '{logger!r}', not provided"
-        if not result_dir:
-            logger.error("Result directory not provided")
-            sys.exit(1)
-        if not target_dir:
-            logger.error("Target directory not provided")
-            sys.exit(1)
-        try:
-            self.result_dir = Path(result_dir).resolve(strict=True)
-        except FileNotFoundError:
-            logger.error("Invalid result directory provided: {}", result_dir)
-            sys.exit(1)
-        else:
-            if not self.result_dir.is_dir():
-                logger.error("Invalid result directory provided: {}", result_dir)
-                sys.exit(1)
-        try:
-            self.target_dir = Path(target_dir).resolve(strict=True)
-        except FileNotFoundError:
-            logger.error("Invalid target directory provided: {}", target_dir)
-            sys.exit(1)
-        else:
-            if not self.target_dir.is_dir():
-                logger.error("Invalid target directory provided: {}", target_dir)
-                sys.exit(1)
+        self.result_dir = result_dir
+        self.target_dir = target_dir
         self.user = user
         self.prefix = prefix
         self.config = config
         self.logger = logger
 
+    def check_result_target_dir(self, result_dir, target_dir):
+        if not result_dir:
+            raise FileNotFoundError("Result directory not provided")
+        else:
+            try:
+                self.result_dir = Path(result_dir).resolve(strict=True)
+            except Exception:
+                raise FileNotFoundError(
+                    "Invalid result directory provided: {}", result_dir
+                )
+            else:
+                if not self.result_dir.is_dir():
+                    self.logger.error(
+                        "Invalid result directory provided: {}", result_dir
+                    )
+                    return 1
+
+        if not target_dir:
+            raise FileNotFoundError("Target directory not provided")
+        else:
+            try:
+                self.target_dir = Path(target_dir).resolve(strict=True)
+            except Exception:
+                raise FileNotFoundError(
+                    "Invalid target directory provided: {}", target_dir
+                )
+            else:
+                if not self.target_dir.is_dir():
+                    self.logger.error(
+                        "Invalid target directory provided: {}", target_dir
+                    )
+                    return 1
```

Comment:
This looks nearly identical to lines 28-42...consider refactoring them into a helper function.  Alternatively, you could turn _this_ function into that helper function and then have the caller validate each of the directories separately.  (That is, if there's no relationship between `result_dir` and `target_dir`, then there's no particular need to have a single function which validates _both_ of them, and you can instead have the caller validate _each_ of them, separately, using (substantially) the same code.)

Reply:
yeah.. done